app/src/main/java/ar/com/wolox/android/cookbook/notifications/helper/NotificationChannelFactory.kt

```diff
@@ -11,32 +11,29 @@ import javax.inject.Inject
 
 @ApplicationScope
 @RequiresApi(Build.VERSION_CODES.O)
-class NotificationChannelFactory @Inject constructor() {
+class NotificationChannelFactory @Inject constructor(@ApplicationScope val context: Context) {
 
     private val notificationChannels: MutableList<NotificationChannel> = arrayListOf()
 
     fun createChannel(channelInfo: NotificationChannelInfo): NotificationChannel =
-        NotificationChannel(channelInfo.id, channelInfo.name, channelInfo.priority)
-            .apply {
+        NotificationChannel(channelInfo.id, channelInfo.name, channelInfo.priority).apply {
                 this.description = channelInfo.description
             }
 
-    fun init(context: Context) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            val generalChannel = createChannel(
-                NotificationChannelInfo(
-                    context.getString(R.string.notifications_general_channel_id),
-                    context.getString(R.string.notifications_general_channel),
-                    context.getString(R.string.notifications_general_channel_description),
-                    NotificationManager.IMPORTANCE_DEFAULT
-                )
+    fun init() {
```

Comment:
This could be done on the `init { ... }` and remove the necessary call on the Application.